data-node/src/main/java/org/graylog/datanode/management/ManagableProcess.java

```diff
@@ -21,7 +21,9 @@
 
 import java.io.IOException;
 
-public interface ManagableProcess {
+public interface ManagableProcess<ConfigurationType> {
+
+    void startWithConfig(ConfigurationType configuration);
 
     void start() throws IOException;
```

Comment:
Please correct me if I am wrong, but it seems that this method will always fail if executed without executing `startWithConfig` first. Shouldn't we name it differently, i.e. `restart`, so that no one tries to execute it too early?

Reply:
good idea, renamed!